File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("RoomChannel", {
  // 接続時
  connected() {
    // Called when the subscription is ready for use on the server
  },

  // 切断時
  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  // サーバーからデータを受信した時
  received(data) {
    alert(data['message']);
  },

  speak: function(message) {
    // サーバーサイドのspeakメソッドをWebSocket経由で呼び出す
    return this.perform('speak', { message: message });
  }
});

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  # 購読後に呼ばれる
  def subscribed
    stream_from "room_channel"
  end

  # 購読解除後に呼ばれる
  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  # クライアントサイドから呼び出された時に実行される
  def speak(data)
    ActionCable.server.broadcast(
      "room_channel", { message: data["message"] }
    )
  end
end
